extensions/mssql/src/connectionconfig/connectionDialogWebviewController.ts

```diff
@@ -499,6 +499,16 @@ export class ConnectionDialogWebviewController extends FormWebviewController<
                 TelemetryActions.LoadFromConnectionString,
             );
 
+            // Helper function to set error message in the appropriate place
+            const setConnectionStringError = (errorMessage: string) => {
+                if (state.dialog?.type === "loadFromConnectionString") {
+                    (state.dialog as ConnectionStringDialogProps).connectionStringError =
+                        errorMessage;
+                } else {
+                    state.formMessage = { message: errorMessage };
+                }
+            };
+
             try {
                 const connDetails =
                     await this._mainController.connectionManager.parseConnectionString(
```

Comment:
Does this actually throw an error if the authentication type isn't one of "AzureMFA", "SqlLogin", or "Integrated"?  I thought the STS PR removes the code that forces it into one of these valid types, so we'll need to do that up here instead.

Reply:
We'll need to add a test that confirms this, too